src/wp-includes/class-wp-query.php

```diff
@@ -941,8 +941,13 @@ public function parse_query( $query = '' ) {
 				$this->is_author = true;
 			}
 
-			if ( ! empty( $qv['post_type'] ) && ! is_array( $qv['post_type'] ) ) {
-				$post_type_obj = get_post_type_object( $qv['post_type'] );
+			if ( ! empty( $qv['post_type'] ) ) {
+				if ( is_array( $qv['post_type'] ) ) {
+					$post_type_obj = get_post_type_object( reset( $qv['post_type'] ) );
+				} else {
+					$post_type_obj = get_post_type_object( $qv['post_type'] );
+				}
+
```

Comment:
I don't understand the purpose for this change.

If an array of multiple post types is passed, then using which ever happens to be first to determine `is_post_type_archive` seems a little bug prone. 

Obviously this change does make sense of an array of one.

Reply:
- The purpose of the change is to make `WP_Query` correctly identify a post type archive when supplied with an array of post types.

- The implementation assumes that if the first post type has an archive, then `WP_Query` should set `is_post_type_archive` to `true`. 

This is in line with how `feed_links_extra()` handles an array of post types:
```php
} elseif ( is_post_type_archive() ) {
	$post_type = get_query_var( 'post_type' );
	if ( is_array( $post_type ) ) {
		$post_type = reset( $post_type );
	}
```
That is, it only respects the first element. [See in source](https://github.com/WordPress/wordpress-develop/blob/0fb4f900304c712cf7a7626cd069ea1b90bda99e/src/wp-includes/general-template.php#L3163).

- The reason the change is necessary is because `is_array( $post_type )` never passes because `is_post_type_archive()` never returns `true` when an array of post types is passed to `WP_Query`.

Reply:
Thanks for the explanation.

That `WP_Query::is_post_type_archive() === false` when multiple post types are queried seems most correct to me. The name suggests a singular focus rather than the plural.

Have you tested what happens if a front end user puts in http://wp-dev.local/?post_type[]=page&post_type[]=some-cpt-with-archives, do both post types get queried?